### DISCUSSION TITLE
Possible feature request/idea - support sorting by file property value?

### DISCUSSION THREAD

**Discussion by windoverwater** [SebastianMC/obsidian-custom-sort#136] [Ideas]
Hi SebastianMC,

First, thanks for this plugin - FWIIW I find it key for my particular usage of obsidian.

Second, the feature request is to be able to sort by the value of a specific file property key.  There are a few pieces of automation that I use which generate various key-value pairs as well as generating files (notes).  Obsidian is a nice tool to both view and integrate that data with manually entered data.  A few of the fields happen to be time (computer epoch time to be precise - no time zone issues).  Being able to sort on those values and other values would be very helpful.

Would it be possible to add support for sorting on the values of specific keys in the file properties?  Thanks!

**Comment by SebastianMC** [comment]
Hi @windoverwater 

I'm also working with automatically created and/or updated notes (and/or their metadata), for which sorting by properties is useful.

This feature has been available for a long time and recently improved with multi-level sorting support. Yet, the documentations is not up-to-date in this area :-|

The tokens (syntax) for sorting by metadata (that's how the properties are referred to) are:
- `with-metadata:` - allows grouping of items with specified metadata
- `by-metadata:` - allows ordering by the value of specified metadata. Multi-level sorting supported

Below an extreme example to show the idea, you can take a subset of it and transform to apply to the specifics of your vault(s):

```yaml
---
sorting-spec: |-
  //     By default in vault root
  //     files sorted by the property 'auto-import-timestamp',
  //     (larger value = newer timestamp -> goes first)
  //     and within the same timestamp by the 'full-name' property
  //     Files w/o the metadata put below these w/ metadata
  //     and sorted (implicitly) alphabetically with files preferred
  //     when a note and a folder share the same (base)name
  target-folder: /
  > a-z by-metadata: auto-import-timestamp, < true a-z by-metadata: full-name

  //     At the same time push to the top files with the property 'has-priority'
  //     and within the group order by the value of the 'has-priority'
  //           (0 = the highest priority)
  //     and if two notes share the same priority value, 
  //         order by note name reverse alph order
  with-metadata: has-priority
    < a-z by-metadata: has-priority, > a-z
---
```

An example vault:

![136-example-vault](https://github.com/SebastianMC/obsidian-custom-sort/assets/23032356/36a470b4-4d39-4ffd-bf86-1f3db64a5489)

Let me know if you need more specific hints

**Reply by windoverwater** [reply]
Thanks for this!  (missed that documentation completely)  Works great!

**Reply by angadsri** [reply]
Awesome!

For the less-technically inclined (such as myself) just posting my implementation too if it can be of any help. 

<img width="308" alt="image" src="https://github.com/user-attachments/assets/f3fa9ed5-23f6-4806-9546-4115476e2312" />

This is for a folder containing my 'source notes' for things I read, watch, etc. — and each gets an "xp dates" property which can contain a list of dates when I experienced/re-experienced the thing. 

(Peculiarity with this case: yes, because the xp dates are a list, the sorting works based on the first date in the list, so it's important to keep the first one the newest date if it's most salient for you to see it in the sidebar ordered by recency.)
